src/SOS/debugshim/debugshim.cpp

```diff
@@ -422,193 +422,196 @@ HRESULT CLRDebuggingImpl::GetCLRInfo(ICorDebugDataTarget* pDataTarget,
+            DWORD fixedFileInfoRVA = ((versionResourceRVA + 3*2 + 16*2 + 3)/4)*4;
+            hr = ReadFromDataTarget(pDataTarget, moduleBaseAddress + fixedFileInfoRVA, (BYTE*)&fixedFileInfo, sizeof(fixedFileInfo));
+        }
 
-    // Record the version information
-    if(SUCCEEDED(hr))
-    {
-        pVersion->wMajor = (WORD) (fixedFileInfo.dwProductVersionMS >> 16);
-        pVersion->wMinor = (WORD) (fixedFileInfo.dwProductVersionMS & 0xFFFF);
-        pVersion->wBuild = (WORD) (fixedFileInfo.dwProductVersionLS >> 16);
-        pVersion->wRevision = (WORD) (fixedFileInfo.dwProductVersionLS & 0xFFFF);
-    }
+        //Verify the signature on the version resource
+        if(SUCCEEDED(hr) && fixedFileInfo.dwSignature != PE_FIXEDFILEINFO_SIGNATURE)
+        {
+            hr = CORDBG_E_NOT_CLR;
+        }
 
-    // Now grab the special clr debug info resource
-    // We may need to scan a few different names searching though...
-    // 1) CLRDEBUGINFO<host_os><host_arch> where host_os = 'WINDOWS' or 'CORESYS' and host_arch = 'X86' or 'ARM' or 'AMD64'
-    // 2) For back-compat if the host os is windows and the host architecture matches the target then CLRDEBUGINFO is used with no suffix.
-    DWORD debugResourceRVA = 0;
-    DWORD debugResourceSize = 0;
-    BOOL useCrossPlatformNaming = FALSE;
-    if(SUCCEEDED(hr))
-    {
-        // the initial state is that we haven't found a proper resource
-        HRESULT hrGetResource = E_FAIL; 
+        // Record the version information
+        if(SUCCEEDED(hr))
+        {
+            pVersion->wMajor = (WORD) (fixedFileInfo.dwProductVersionMS >> 16);
+            pVersion->wMinor = (WORD) (fixedFileInfo.dwProductVersionMS & 0xFFFF);
+            pVersion->wBuild = (WORD) (fixedFileInfo.dwProductVersionLS >> 16);
+            pVersion->wRevision = (WORD) (fixedFileInfo.dwProductVersionLS & 0xFFFF);
+        }
+
+        // Now grab the special clr debug info resource
+        // We may need to scan a few different names searching though...
+        // 1) CLRDEBUGINFO<host_os><host_arch> where host_os = 'WINDOWS' or 'CORESYS' and host_arch = 'X86' or 'ARM' or 'AMD64'
+        // 2) For back-compat if the host os is windows and the host architecture matches the target then CLRDEBUGINFO is used with no suffix.
+        DWORD debugResourceRVA = 0;
+        DWORD debugResourceSize = 0;
+        BOOL useCrossPlatformNaming = FALSE;
+        if(SUCCEEDED(hr))
+        {
+            // the initial state is that we haven't found a proper resource
+            HRESULT hrGetResource = E_FAIL; 
      
-        // First check for the resource which has type = RC_DATA = 10, name = "CLRDEBUGINFO<host_os><host_arch>", language = 0        
-#if defined (HOST_IS_WINDOWS_OS) && defined(_HOST_X86_)
-        const WCHAR * resourceName = W("CLRDEBUGINFOWINDOWSX86");
-#endif
+            // First check for the resource which has type = RC_DATA = 10, name = "CLRDEBUGINFO<host_os><host_arch>", language = 0        
```

Comment:
Are these tabs? They need to be spaces. This applies to this whole file.

Reply:
Probably my VS preferences are wrong.

Reply:
> GetMachineAndResourceSectionRVA is in the dbgutil library and ifdef'ed FEATURE_PAL

I would have expected no forward declaration warning as error to stop compile.  Maybe we could turn on something like that???

Reply:
They are not tabs.  I have view whitespace turned on in VS....  I did ident the whole block.  Maybe you would prefer I didn't ident the whole thing.

Reply:
They looked strange in the diff but if they are spaces I'm fine.

Reply:
Looks like I forgot to ifdef the declarations in inc\dbgshim.h.